core/src/main/scala/com/salesforce/op/filters/FeatureDistribution.scala

```diff
@@ -179,40 +179,46 @@ private[op] object FeatureDistribution {
 
   /**
    * Function to put data into histogram of counts
-   * @param values values to bin
-   * @param sum summary info for feature (max and min)
-   * @param bins number of bins to produce
-   * @param hasher hasing function to use for text
+   *
+   * @param values  values to bin
+   * @param summary summary info for feature (max, min, etc)
+   * @param bins    number of bins to produce
+   * @param textBinsFormula formula to compute the text features bin size
    * @return the bin information and the binned counts
    */
-  // TODO avoid wrapping and unwrapping??
   private def histValues(
     values: ProcessedSeq,
-    sum: Summary,
+    summary: Summary,
     bins: Int,
-    hasher: HashingTF
-  ): (Array[Double], Array[Double]) = {
-    values match {
-      case Left(seq) => Array(sum.min, sum.max) -> hasher.transform(seq).toArray // TODO use summary info to pick hashes
-      case Right(seq) => // TODO use kernel fit instead of histogram
-        if (sum == Summary.empty) {
-          Array(sum.min, sum.max) -> seq.toArray // the seq will always be empty in this case
-        } else if (sum.min < sum.max) {
-          val step = (sum.max - sum.min) / (bins - 2.0) // total number of bins includes one for edge and one for other
-          val splits = (0 until bins).map(b => sum.min + step * b).toArray
-          val binned = seq.map { v =>
-            NumericBucketizer.bucketize(
-              splits = splits, trackNulls = false, trackInvalid = true,
-              splitInclusion = Inclusion.Left, input = Option(v)
-            ).toArray
-          }
-          val hist = binned.fold(new Array[Double](bins))(_ + _)
-          splits -> hist
-        } else {
-          val same = seq.map(v => if (v == sum.max) 1.0 else 0.0).sum
-          val other = seq.map(v => if (v != sum.max) 1.0 else 0.0).sum
-          Array(sum.min, sum.max) -> Array(same, other)
+    textBinsFormula: (Summary, Int) => Int
+  ): (Array[Double], Array[Double]) = values match {
+    case Left(seq) =>
+      val numBins = textBinsFormula(summary, bins)
```

Comment:
why would the formula need the bins?

Reply:
Right now we set is as `def textBinsFormula(s: Summary, bins: Int) = bins`.
I can image default `bins` can be returned in some cases.

Reply:
what would you recommend then? @leahmcguire

Reply:
bins is the default number (e.g. 100) there might be cases that we will just want to fall back to the defaults